### PR TITLE
fix: use custom repository via DIRIGIBLE_REPOSITORY_LOCAL_ROOT_FOLDER

### PR DESCRIPTION
**Pull request by iliyan-velichkov**

### CODE REVIEW

**Comment by iliyan-velichkov** [comment]
Preventive measure for this issue: https://github.com/eclipse/dirigible/issues/3175

**Comment by iliyan-velichkov** [comment]
@ThuF @delchev @vmutafov the pr is for review

File: components/core/core-repository/src/main/java/org/eclipse/dirigible/components/repository/RepositoryConfig.java [hunk 1]
```diff
@@ -10,20 +10,26 @@
...
-@Configuration
+@org.springframework.context.annotation.Configuration
 public class RepositoryConfig {
 
+
     @Bean
     public IRepository repository() {
-        LocalRepository localRepository = new LocalRepository("target");
+        String repoFolderPath = Configuration.get(Configuration.DIRIGIBLE_REPOSITORY_LOCAL_ROOT_FOLDER, "target");
+        boolean absolutePath = repoFolderPath.startsWith(File.separator);
```
Comment (delchev) [inline components/core/core-repository/src/main/java/org/eclipse/dirigible/components/repository/RepositoryConfig.java:27]: let use the explicit DIRIGIBLE_REPOSITORY_LOCAL_ROOT_FOLDER_IS_ABSOLUTE here as to work on any filesystem

**Review by delchev — CHANGES_REQUESTED** [review]

File: components/core/core-repository/src/main/java/org/eclipse/dirigible/components/repository/RepositoryConfig.java (hunk 1 above)
Comment (iliyan-velichkov) [reply components/core/core-repository/src/main/java/org/eclipse/dirigible/components/repository/RepositoryConfig.java:27]: It uses the files separator to check whether it is absolute.  This way it should be working on every os or I'm missing something?

Comment (delchev) [reply components/core/core-repository/src/main/java/org/eclipse/dirigible/components/repository/RepositoryConfig.java:27]: on Windows you could have something like `C:\MyFolder` which will not be taken as an absolute

Comment (iliyan-velichkov) [reply components/core/core-repository/src/main/java/org/eclipse/dirigible/components/repository/RepositoryConfig.java:27]: Somehow I forgot about the Windows :)
I will fix it on Monday.

**Review by delchev — APPROVED** [review]

**Pull request merged**